src/display/node_stream.js

```diff
@@ -288,8 +260,29 @@ class PDFNodeStreamRangeReader extends BaseRangeReader {
     this._request.end();
   }
 
+  read() {
+    return this._readCapability.promise.then(() => {
+      if (this._done) {
+        return Promise.resolve({ value: undefined, done: true, });
+      }
+      if (this._errored) {
+        return Promise.reject(this._reason);
+      }
+
+      let chunk = this._rangeRequest.read();
```

Comment:
The context is missing from the comment that you've linked. I think that the breakage is unrelated to the method being in the base class. You now literally have two (actually, after looking again THREE, `BaseFullReader.prototype.read` is almost identical, except with the "length" property in `onProgress`) identical methods `read` in the sub classes.

The inheritance tree looks like this:

- BaseRangeReader
- PDFNodeStreamRangeReader extends BaseRangeReader
- PDFNodeStreamFsRangeReader extends BaseRangeReader

Your current code duplicates the `read` method in the latter two classes (the implementation is byte for byte identical). The runtime behavior is identical regardless of where you put the implementation of the "read" method, so you should put the implementation in the base class (BaseRangeReader), to improve the maintainability of the code.